sky/engine/core/text/Paragraph.cpp

```diff
@@ -19,46 +21,51 @@ Paragraph::~Paragraph()
 
 double Paragraph::width()
 {
-    return m_renderView->firstChildBox()->width();
+    return firstChildBox()->width();
 }
 
 double Paragraph::height()
 {
-    return m_renderView->firstChildBox()->height();
+    return firstChildBox()->height();
 }
 
 double Paragraph::minIntrinsicWidth()
 {
-    return 0.0;
+    return firstChildBox()->minPreferredLogicalWidth();
 }
 
 double Paragraph::maxIntrinsicWidth()
 {
-    return 0.0;
+    return firstChildBox()->maxPreferredLogicalWidth();
 }
 
 double Paragraph::alphabeticBaseline()
 {
-    return 0.0;
+    return firstChildBox()->firstLineBoxBaseline(FontBaselineOrAuto(AlphabeticBaseline));
 }
 
 double Paragraph::ideographicBaseline()
 {
-    return 0.0;
+    return firstChildBox()->firstLineBoxBaseline(FontBaselineOrAuto(IdeographicBaseline));
 }
 
 void Paragraph::layout()
 {
     LayoutUnit maxWidth = std::max(m_minWidth, m_maxWidth);
     LayoutUnit maxHeight = std::max(m_minHeight, m_maxHeight);
-    IntSize maxSize(maxWidth, maxHeight);
-
-    m_renderView->setFrameViewSize(maxSize);
+    m_renderView->setFrameViewSize(IntSize(maxWidth, maxHeight));
     m_renderView->layout();
 }
 
 void Paragraph::paint(Canvas* canvas, const Offset& offset)
 {
+    // Very simplified painting to allow painting an arbitrary (layer-less) subtree.
+    GraphicsContext context(canvas->skCanvas());
+    Vector<RenderBox*> layers;
+    PaintInfo paintInfo(&context, m_renderView->absoluteBoundingBoxRect(), m_renderView.get());
+    LayoutPoint paintOffset(offset.sk_size.width(), offset.sk_size.height());
+    m_renderView->RenderBox::paint(paintInfo, paintOffset, layers);
+    // Note we're ignoring any layers encountered.
```

Comment:
add a todo or file an issue or something on what needs doing here just in case someone else has to take over